### ISSUE TITLE
Write the script we wish we could run

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/clmassmod#3]
[This](https://github.com/LSSTDESC/clmassmod/blob/refactoring/docs/wishlist.ipynb) will guide how we redesign the code.

**Comment by aimalz** [comment]
After working through more of the existing code with @cavestruz today, I'm assigning myself to write up our questions about what parts are of interest to reimplement and which things we might want to change entirely.

**Comment by tmcclintock** [comment]
In an ideal world someone could do the following:
```python
import clmassmod

#Pretend we have a M_obs from a lensing analysis somewhere
mass_bias = clmassmod.get_mass_bias(M_obs)
```
Where `mass_bias` is either a fractional change to M_obs or a total change to M_obs that needs to be applied. There are many reasons why it won't be this easy, but this is the ideal.

**Comment by aimalz** [comment]
@tmcclintock That's basically what we've got [here](https://github.com/LSSTDESC/clmassmod/blob/refactoring/docs/wishlist.ipynb).  Note, however, that your version takes only `M_obs` as an argument.  What we discovered earlier today was that @deapplegate's version requires `M_true` as an additional input because it does not perform analysis of masses but instead only of the mass bias.  I was hoping that my contributions to this project could be run on mock data to establish bias and then on real data where the truth is not known, consistent with your proposed API.  However, I misunderstood the model, and it now looks like this functionality could require significant work beyond a simple refactoring.

**Comment by tmcclintock** [comment]
I agree. Some of the reasons I alluded to about this being so difficult to do in general are that the mass bias

- depends on the model you use 
- depends on the covariance you use, i.e. are you using a covariance estimated from the sims or does it come from your data?
- depends on which sims you use
- depends on how you define halos

Making a tool to provide a mass bias is possible, but to make it fully general is very difficult.

**Comment by aimalz** [comment]
Re: mass bins, one change we're suggesting is for mass bins to be user defined, not built in, so those choices would be made by the user before they call any perform any analysis with clmassmod.  This is something I insist on with all my codes, both because I anticipate that binning will become obsolete as analysis methods advance and we can work in the continuous domain and because I'm stubborn about ensuring that something aiming to recover the truth must never know the truth a priori.

Re: bias, the suggested change would separate the fitting of a global (or per bin) mass distribution and the estimation of a mass bias.  But, that's not actually something the existing code can do, from what we can tell.
